README.md

```diff
@@ -342,20 +349,43 @@ for hiredis-cluster as well.
 
 ### Connecting
 
-The function `redisClusterAsyncConnect` can be used to
-establish a set of non-blocking connections to a Redis cluster.
-It returns a pointer to the newly created `redisClusterAsyncContext` struct.
-The `err` field should be checked after creation
-to see if there were errors creating the asynchronous cluster context.
+There are two alternative ways to initiate a cluster client which also determines
+how the client behaves during the initial connect.
+
+The first alternative is to use the function `redisClusterAsyncConnect`, which initially
+connects to the cluster in a blocking fashion and waits for the slot map before returning.
+Any command sent by the user thereafter will create a new non-blocking connection,
+unless a non-blocking connection already exists to the destination.
+The function returns a pointer to a newly created `redisClusterAsyncContext` struct.
+The `err` field should be checked after creation to make sure the slot map could be
+fetched correctly.
 
 ```c
 redisClusterAsyncContext *acc = redisClusterAsyncConnect("127.0.0.1:6379", HIRCLUSTER_FLAG_NULL);
 if (acc->err) {
     printf("Error: %s\n", acc->errstr);
-    // handle error
+    // handle error..
 }
+// ..or attach an event engine
+```
+
+The second alternative is to use `redisClusterAsyncContextInit` and `redisClusterAsyncConnect2`
+which avoids the initial blocking connect. This connection alternative requires an attached
+event engine when called but the initial connect and fetch of slot map is done in a
+non-blocking fashion.
+
+```c
+redisClusterAsyncContext *acc = redisClusterAsyncContextInit();
```

Comment:
```suggestion
// Error handling omitted for brevity.
redisClusterAsyncContext *acc = redisClusterAsyncContextInit();
```

Otherwise we would need to check acc == NULL too, and base == NULL too...

Reply:
Btw, we need to add some nodes too before calling connect2 in this example. Is `redisClusterSetOptionAddNodes(acc->cc, ...)` the way to go or is there a better way?